src/panels/LogPanel/ScriptLogPanel.tsx

```diff
@@ -31,46 +37,133 @@ export function ScriptLogPanel() {
+          return script;
+        })
     );
+
+    setSelectedScripts(new Set());
   }, [luaApi]);
 
   useEffect(() => {
     fetchScriptLogEntries();
   }, [fetchScriptLogEntries]);
 
-  function preprocessScriptEntry(entry: string): string {
-    // @TODO (anden88, 2025-04-17): do we want to rewrite this using regex instead?
-    // return entry.replace(/[.,()""]/g, ' ');
-    return entry
-      .replaceAll('.', ' ')
-      .replaceAll(',', ' ')
-      .replaceAll('"', ' ')
-      .replaceAll('(', ' ')
-      .replaceAll(')', ' ');
+  function sanitizeScript(script: string): string {
+    // Replace special characters with spaces. The '+' ensures repeated punctuation is
+    // replaced by a single space instead of many
+    return script.replace(/[.,()[\]{}"'\\/]+/g, ' ');
+  }
+
+  function handleToggleSelection(index: number) {
+    setSelectedScripts((prev) => {
+      const newSet = new Set(prev);
+      if (newSet.has(index)) {
+        newSet.delete(index);
+      } else {
+        newSet.add(index);
+      }
+      return newSet;
+    });
+  }
+
+  function runSelectedScripts() {
+    if (scriptLogEntries === undefined || !luaApi) {
+      return;
+    }
+    // TODO (anden88 2025-04-23): In what order do we want to run the scripts? The scripts
+    // are ordered latest first (top-bottom) but they were actually executed bottom up...
+    const scriptsToRun = Array.from(selectedScripts)
+      .sort((a, b) => a - b)
+      .map((index) => scriptLogEntries[index]);
+
+    scriptsToRun.map((script) => luaApi.scheduleScript(script, 0));
+
+    setSelectedScripts(new Set());
   }
 
   return (
-    <FilterList isLoading={scriptLogEntries.length === 0}>
-      <Group gap={'xs'}>
-        <Tooltip label={'Refresh script log'}>
-          <ActionIcon onClick={fetchScriptLogEntries} aria-label={'Refresh script log'}>
-            <ReplayIcon />
-          </ActionIcon>
-        </Tooltip>
-        <FilterList.InputField placeHolderSearchText={'Search for a script'} flex={1} />
-      </Group>
-      <FilterList.SearchResults
-        data={scriptLogEntries}
-        renderElement={(entry) => <ScriptLogEntry script={entry} />}
-        matcherFunc={(data, searchString) => {
-          const preprocessedData = preprocessScriptEntry(data);
-          return wordBeginningSubString(preprocessedData, searchString);
-        }}
-      >
-        <FilterList.SearchResults.VirtualList gap={'xs'} />
-      </FilterList.SearchResults>
-    </FilterList>
+    <Layout>
+      <Layout.FixedSection mb={'xs'}>
+        <Group wrap={'nowrap'} justify={'space-between'} align={'start'} gap={'xs'}>
+          <Group gap={'xs'}>
+            <Tooltip label={'Run all the selected scripts in sequential order'}>
+              <Button
+                onClick={runSelectedScripts}
+                rightSection={<RerunScriptIcon />}
+                disabled={selectedScripts.size === 0}
+              >
+                Run Scripts
+              </Button>
```

Comment:
Can we make this styled the same as the other run scripts buttons? I.e. green and subtle 🙏